doc/mb_compare.md

```diff
@@ -0,0 +1,34 @@
+# MirrorCache (mc) concept vs MirrorBrain (mb) concept
+
+1. Scanning
+1.1
+mb uses daily full scan of all files on all mirrors, without regard of whether files were ever downloaded or changed recently. This may be waste of resources.
+mc scans only those locations which are needed by users and prioritize the scans according to popularity.
+1.2
+It may take up to 24 hours before mb notices that file has gone from a mirror. Clients will be incorrectly redirected and receive 404 error.
+mc is making sure that file exists on selected mirror and never redirects to locations where file was missing for a while.
+1.3
+New files are added by scanners in mb, which needs careful handling of deadlocks when two scanners attempt to add the same files.
+mc has separate jobs for keeping in sync information about files on the Main Server and for scanning mirrors. In additional, jobs have natural locking mechanism (https://docs.mojolicious.org/Minion#lock), which prevents conflicting operations.
+1.4
+New files are added by scanners in mb, which currently leads to many incorrect files in database. (Potentially overload of mb DB because of incorrectly configured mirror).
+(This may fixed in cost of added coupling on mb architecture).
+
+2. Maintenance jobs, scheduling, logging, resources and scalability.
+2.1
+mb relies on command line tool and cron to run jobs. User then manually reviews logs to find eventual problems (which is a challenge to do with current amount of logging).
+mc has popular mature fast asynchronous job queue https://docs.mojolicious.org/Minion, which includes prioritization, rescheduling, custom job queues, scalability (adding workers), manageability (assigning workers to particular queues), dashboard, search, etc.
+
+3. WWW
+mb relies on rendering of static files generated by cron job.
+mc has popular mature fast asynchronous framework with delayed rendering, javascript, etc. (mojolicious.org)
+
+4. Fileless.
+4.1
+The roadmap of mb has fileless architecture, which relies on empty files and default index rendering by Apache. Additionally to extra filesystem's requirements, this looks rather as a hack and source of inconsistency (For each file on the main server there must be empty file in mb and a record in DB).
```

Comment:
the 0 byte file mode is already there. the roadmap wants to go file less without the 0 byte files.